packages/contracts-bedrock/hardhat.config.ts

```diff
@@ -10,6 +10,11 @@ import 'hardhat-deploy'
 // Hardhat tasks
 import './tasks'
 
+let bytecodeHash = 'none'
+if (process.env.FOUNDRY_PROFILE === 'echidna') {
+  bytecodeHash = 'ipfs'
+}
+
```

Comment:
Echidna needs the metadata hashes to work properly. The `FOUNDRY_PROFILE` flag is natively recognized by the `foundry.toml` file, and it's been manually added to have the same effect in the hardhat config file.